File: assignment-04/Task 02/c/main.c
#include "str_format.h"
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {

  // Declares strings
  const char *str1 = "x & y";
  const char *str2 = "x < y & y > 0";
  const char *str3 = "& test < 123 <> &&";

  // Formats strings
  char *out1 = strFormat(str1, strlen(str1));
  char *out2 = strFormat(str2, strlen(str2));
  char *out3 = strFormat(str3, strlen(str3));

  // Prints strings
  printf("%-18s -> %s\n", str1, out1);
  printf("%-18s -> %s\n", str2, out2);
  printf("%-18s -> %s\n", str3, out3);

  // Frees pointers
  free(out1);
  free(out2);
  free(out3);

  return 0;
}

File: assignment-04/Task 02/rust/main.rs
fn main() {
    let str1 = String::from("x & y");
    let str2 = String::from("x < y & y > 0");
    let str3 = String::from("& test < 123 <> &&");
    
    println!("{: <18} -> {}", "x & y", str_format(str1));
    println!("{: <18} -> {}", "x < y & y > 0", str_format(str2));
    println!("{: <18} -> {}", "& test < 123 <> &&", str_format(str3));
}

fn str_format(input: String) -> String {
    return input.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;");     
}
